### ISSUE TITLE
Hyperparameters for Multi-GPU training

### ISSUE THREAD

**Issue by LMdeLiangMi** [Wangt-CN/VC-R-CNN#4]
Hi, @Wangt-CN 

I use the following commands to perform multi-gpu training:
```
export NGPUS=4
CUDA_VISIBLE_DEVICES=2,3,4,5 python -m torch.distributed.launch --nproc_per_node=$NGPUS tools/train_net.py --config-file "configs/e2e_mask_rcnn_R_101_FPN_1x.yaml" MODEL.RPN.FPN_POST_NMS_TOP_N_TRAIN 2000
```

Should I add the hyperparameters from the single-gpu training command? 

`python tools/train_net.py --config-file "configs/e2e_mask_rcnn_R_101_FPN_1x.yaml" SOLVER.IMS_PER_BATCH 2 SOLVER.BASE_LR 0.0025 SOLVER.MAX_ITER 720000 SOLVER.STEPS "(480000, 640000)" TEST.IMS_PER_BATCH 1 MODEL.RPN.FPN_POST_NMS_TOP_N_TRAIN 2000
`

**Comment by Wangt-CN** [comment]
Hi, it is seemed that you are using 4 GPUs for training. You needn't add the whole hyperparameters from the single-gpu training. However, maybe you need to change the learning rate and the steps. For which, you can just change them in the config file directly, for example `configs/e2e_mask_rcnn_R_101_FPN_1x.yaml`: https://github.com/Wangt-CN/VC-R-CNN/blob/master/configs/e2e_mask_rcnn_R_101_FPN_1x.yaml#L37

For other parameters, you need not change them in the bash unless you use other number of GPUs.
**If you change the number of GPUs**, you should change `SOLVER.IMS_PER_BATCH` (the default in my config is 8, i.e., for 4 GPUs)

**Comment by LMdeLiangMi** [comment]
Hi, @Wangt-CN  Thanks.

![image](https://user-images.githubusercontent.com/13929748/77381614-57583f80-6db9-11ea-90ed-0d8c05180ffb.png)

Referring to the paper's 5.2, I may need to change configs/e2e_mask_rcnn_R_101_FPN_1x.yaml as following to reproduce your result? What about the **STEPS** and **Max_ITER**?
```
SOLVER:
  BASE_LR: 0.0002
  WEIGHT_DECAY: 0.0001
  STEPS: (160000, 200000)
  MAX_ITER: 240000
```

**Comment by LMdeLiangMi** [comment]
Hi, @Wangt-CN  Thanks!

![image](https://user-images.githubusercontent.com/13929748/77386302-778dfb80-6dc5-11ea-8a9f-21dc507e22d5.png)
Referring to the paper's 5.2, should I change the configs/e2e_mask_rcnn_R_101_FPN_1x.yaml as following to reproduce your result? What about **STEPS**?
```
SOLVER:
  BASE_LR: 0.0002
  WEIGHT_DECAY: 0.0001
  **STEPS: (160000, 200000)**
  MAX_ITER: 180000
```

**Comment by Wangt-CN** [comment]
Hi, actually you can just follow our current config file in repo (For MSCOCO Dataset and our structure). That means if you change any model or dataset you may change the hyperparameters.
The lr has changed from 0.0002 to 0.0005 since in our further experiments we found lr = 0.0005 can be a little better (the loss can decrease lower). For the other hyperparameters, it is not very important actually and you can just follow our current config file.   

Moreover, we have also discussed in our repo that it can be hard to find the exactly best lr and hyperparameters actually, since it needs to test the results in the downstream tasks.  You can also just have another try, especially for different proxy tasks or structures.
